lib/ripper-tags.rb

```diff
@@ -136,15 +140,35 @@ def self.option_parser(options)
 
   def self.process_args(argv, run = method(:run))
     option_parser(default_options) do |optparse, options|
-      file_list = optparse.parse(argv)
+      invalid_options = []
+      file_list = []
+
+      while argv.size > 0
```

Comment:
At each failure, we remove the offending option and try again. As long as optparse doesn't work destructively (itself removing things from argv), this works nicely. Halt when parsing succeeds (using `break`), or no (valid) arguments left.